lib/my_john_deere_api/net_http_retry/invalid_response_error.rb

```diff
@@ -0,0 +1,23 @@
+module MyJohnDeereApi
+  module NetHttpRetry
+    ##
+    # This error is used when a single request has exceeded
+    # the number of retries allowed by NetHttpRetry::Decorator::MAX_RETRIES.
+
+    class InvalidResponseError < StandardError
+
+      ##
+      # argument is a string which describes the attempted request
+
+      def initialize(response)
+        message = {
+          code: response.code,
+          message: response.message,
```

Comment:
Looking at the logs in our application JD error messages look like this:
```
 Received fault from next hop: faultcode : [409] faultstring : [] 
```
So maybe we can create a mapping of ones that we know/expect to something more meaningful?

Reply:
see my note above :)